### ISSUE TITLE
🔗 integration test coverage for grpc-web compatibility

### ISSUE THREAD

**Issue by cratelyn** [penumbra-zone/penumbra#3704]
it is an important requirement of `pd` that we have maintain compatibility with gRPC-web clients. the https://github.com/penumbra-zone/web project depends upon [`connect-es`](https://github.com/connectrpc/connect-es/) to communicate with nodes via the gRPC-web protocol.

to address #3627, we tested that #3652 maintained compatibility by spinning up a live validator and connecting to it via the web wallet. this is a time-consuming process that we can't perform on every PR, so we nonetheless observed a regression after landing other seemingly innocuous changes to `pd`'s router. (_see #3697, #3679_)

we should introduce some integration test coverage that demonstrates that a gRPC-web client can properly communicate with a fullnode.

i would prefer if the solution to this does not entail depending on the web repository as a submodule. perhaps we could spin up a test client that also uses the same `connect-es` library, to show that transport works properly.

**Comment by hdevalence** [comment]
Should this be considered a dup of https://github.com/penumbra-zone/web/issues/415 ?

**Comment by cratelyn** [comment]
> Should this be considered a dup of [penumbra-zone/web#415](https://github.com/penumbra-zone/web/issues/415) ?

i don't believe so. that is discussing the web project's CI; i'd like if we had something in the penumbra monorepo to exercise this.

i'm not privy to how they are planning on wiring up playwright though, presumably it would be running against the preview network? in that case, we would still be in a position where our test suite would not catch regressions that could break the web extension until after changes have been merged.

i'd like something that, while perhaps not exhaustive, gives us some confidence that grpc-web clients can talk to pd. does that sound fair to you, @hdevalence?

**Comment by hdevalence** [comment]
Yes, the issue is just that we aren't putting any web tooling in this repo, to insulate the core protocol development from the churn of web tooling. So unless we have a Rust grpc-web client handy we'll need to get delayed notifications via a CI run triggered in another repo. I think that's fine, since those breaks should be rare — this is the first one in more than a year.

**Comment by conorsch** [comment]
Not having this coverage in place was painful: https://github.com/penumbra-zone/penumbra/pull/4351
